### ISSUE TITLE
Cannot build lighttpd:1913-1914

### ISSUE THREAD

**Issue by afsafzal** [squaresLab/ManyBugs#7]
```
building bug: manybugs:lighttpd:1913-1914
Building image: squareslab/manybugs:base
Step 1/12 : FROM ubuntu:14.04

 ---> 2ff3b426bbaa
Step 2/12 : MAINTAINER Chris Timperley "christimperley@gmail.com"

 ---> Using cache
 ---> 1640b372fd57
Step 3/12 : RUN apt-get update &&     apt-get install --no-install-recommends -y sudo &&     useradd -ms /bin/bash docker &&     echo 'docker ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers &&     adduser docker sudo &&     apt-get clean &&     mkdir -p /home/docker &&     sudo chown -R docker /home/docker &&     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*

 ---> Using cache
 ---> fe6c78721856
Step 4/12 : USER docker

 ---> Using cache
 ---> 7c4a7a993dfa
Step 5/12 : RUN sudo chown -R docker /usr/local/bin

 ---> Using cache
 ---> df229f8c11e6
Step 6/12 : RUN sudo apt-get update &&     sudo apt-get install  --no-install-recommends -y                           build-essential                           gcc                           patch                           curl                           libcap-dev                           git                           cmake                           vim                           jq                           tar                           psmisc                           moreutils                           wget                           zip                           unzip                           python3-setuptools                           python                           software-properties-common                           libncurses5-dev &&     sudo apt-get autoremove -y &&     sudo apt-get clean &&     sudo rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*

 ---> Using cache
 ---> a8c3de49f196
Step 7/12 : RUN sudo rm /bin/sh &&     sudo ln -s /bin/bash /bin/sh

 ---> Using cache
 ---> 290c5b222cd4
Step 8/12 : RUN sudo mkdir -p /experiment && sudo chown -R docker /experiment

 ---> Using cache
 ---> 7346f3f7a4fd
Step 9/12 : WORKDIR /experiment

 ---> Using cache
 ---> e7580fca529e
Step 10/12 : ADD base/preprocess /experiment/preprocess

 ---> Using cache
 ---> 940cf19e2006
Step 11/12 : ADD compile.sh /experiment/compile.sh

 ---> Using cache
 ---> 0fb211ba3a47
Step 12/12 : RUN sudo chown -R docker /experiment &&     sudo chmod +x compile.sh

 ---> Using cache
 ---> 0409eaf9ce6c
Successfully built 0409eaf9ce6c
Successfully tagged squareslab/manybugs:base
Built image: squareslab/manybugs:base
Building image: squareslab/manybugs:lighttpd
Step 1/3 : FROM squareslab/manybugs:base

 ---> 0409eaf9ce6c
Step 2/3 : MAINTAINER Chris Timperley "christimperley@gmail.com"

 ---> Using cache
 ---> c19c0c3cb914
Step 3/3 : RUN sudo apt-get update &&     sudo apt-get install --no-install-recommends -y subversion                                                     m4                                                     libglib2.0-dev

 ---> Using cache
 ---> 2be49294a6b5
Successfully built 2be49294a6b5
Successfully tagged squareslab/manybugs:lighttpd
Built image: squareslab/manybugs:lighttpd
Building image: squareslab/manybugs:lighttpd-1913-1914
Step 1/14 : FROM squareslab/manybugs:lighttpd

 ---> 2be49294a6b5
Step 2/14 : ARG bug_commit

 ---> Using cache
 ---> e94a4ebc6cc3
Step 3/14 : ARG fix_commit

 ---> Using cache
 ---> f16987987561
Step 4/14 : ENV BUG_COMMIT ${bug_commit}

 ---> Using cache
 ---> 7a781b159857
Step 5/14 : ENV FIX_COMMIT ${fix_commit}

 ---> Using cache
 ---> 9e0858eb432d
Step 6/14 : ENV SCENARIO_NAME lighttpd-bug-${BUG_COMMIT}-${FIX_COMMIT}

 ---> Using cache
 ---> 264e0cd7413b
Step 7/14 : ENV OLD_LOCATION /root/mountpoint-genprog/genprog-many-bugs/${SCENARIO_NAME}

 ---> Using cache
 ---> a77366ac7729
Step 8/14 : RUN cd /experiment &&     wget  -nv http://repairbenchmarks.cs.umass.edu/ManyBugs/scenarios/${SCENARIO_NAME}.tar.gz           -O scenario.tar.gz &&     tar -xf scenario.tar.gz &&     mv ${SCENARIO_NAME} scenario &&     rm scenario.tar.gz &&     cd scenario &&     rm -rf  coverage*             configuration-oracle             local-root             limit*             *.cache             *.debug.*             sanity             compile.pl             *~             test             reconfigure             preprocessed             fixed-program.txt &&     mv bug-failures bug-info &&     mv bugged-program.txt manifest.txt &&     mv *.lines bug-info &&     mv fix-failures bug-info &&     mv lighttpd src &&     cd ../ &&     mv scenario/* /experiment &&     rm -rf scenario &&     cd /experiment &&     rm -rf tests &&     cd /experiment/src &&       make distclean &&       svn upgrade &&       svn revert $(cat /experiment/manifest.txt) &&     sudo chown -R docker /experiment

 ---> Using cache
 ---> 79c8ccb3d3f7
Step 9/14 : RUN sed -i "s#${OLD_LOCATION}#/experiment#g" test.sh &&     sed -i "s#cd gmp#cd src#g" test.sh &&     sed -i "s#/experiment/limit#timeout 5#g" test.sh &&     sed -i "s#/usr/bin/perl#perl#g" test.sh &&     sed -i 's#lt-\.\*#lt-\.\* \&\> /dev/null#g' test.sh &&     sed -i 's#cd lighttpd/tests#cd src/tests#g' test.sh

 ---> Using cache
 ---> 50261cc45408
Step 10/14 : RUN sed -i 's#image.JPG#image.jpg#g' src/tests/core-request.t

 ---> Using cache
 ---> 77fe128e4012
Step 11/14 : RUN cd src/tests/tmp/lighttpd/servers/www.example.org/pages &&     rm symlinked index.xhtml &&     ln -s expire symlinked &&     ln -s index.html index.xhtml

 ---> Using cache
 ---> 42113081dc7e
Step 12/14 : ADD mod-cgi.t /experiment/src/tests/mod-cgi.t

 ---> Using cache
 ---> 28dc420e092e
Step 13/14 : RUN ./preprocess &&     cd src &&     ./configure &&     make -j$(nproc)

 ---> Running in 79b91b02cf20
checking build system type...
x86_64-unknown-linux-gnu
checking host system type...

....

FAILS DURING COMPILATION

Traceback (most recent call last):
  File "/home/afsoon/BugZoo/bin/bugzoo", line 9, in <module>
    load_entry_point('bugzoo==1.1.4', 'console_scripts', 'bugzoo')()
  File "/home/afsoon/BugZoo/bugzoo/cli.py", line 524, in main
    args.func(args)
  File "/home/afsoon/BugZoo/bugzoo/cli.py", line 486, in <lambda>
    cmd.set_defaults(func=lambda args: build_bug(rbox, args.bug, args.force))
  File "/home/afsoon/BugZoo/bugzoo/cli.py", line 94, in build_bug
    bug.build(force=force)
  File "/home/afsoon/BugZoo/bugzoo/bug.py", line 309, in build
    self.__build_instructions.build(force=force, quiet=quiet)
  File "/home/afsoon/BugZoo/bugzoo/build.py", line 230, in build
    raise ImageBuildFailed(self.tag, log)
bugzoo.errors.ImageBuildFailed: squareslab/manybugs:lighttpd-1913-1914
```

**Comment by ChrisTimperley** [comment]
Nice catch. The preprocess script uses `make -j$(nproc)` to build, which can produce non-deterministic results. For now, I'm going to change the script to `make -j8`, but I could always add a `--threads` argument to the `preprocess` script.

**Comment by ChrisTimperley** [comment]
Fixed by 118dc26.
